[Home] - 스와이퍼, Tag-canvas 기능 추가

kelvin-chihyun opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- 홈페이지 내 스와이퍼, 태그 캔버스 기능을 추가하였습니다! (초안입니다)

1. Swiper
![스와이퍼](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/8f4dc0b9-398a-4a45-900c-0531665654f4)

2. TagCanvas
![태그캔버스](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/d428c3ff-f4be-4d84-85ab-9635fa033c52)


3. Industry 부분 레이아웃
![이미지 레이아웃-min](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/7357e8ff-e595-42da-87a0-d667735a94df)


## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

-

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

-

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

-

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

-

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

kelvin-chihyun committed:
✨ [Feat] : 스와이퍼 기능 추가

홈(메인) 페이지 - Partner 컴포넌트에 스와이퍼 기능을
추가하였습니다.

Related to #52

kelvin-chihyun committed:
✨ [Feat] : Industry section 라우팅 연결

메인 페이지 industry 섹션에서 Industry 컴포넌트로 라우팅을 연결하였습니다.

Related to #50

kelvin-chihyun committed:
✨ [Design] : 디자인 일부 수정

Industries 부분의 디자인 시안이 변경되어,
기반하여 레이아웃을 변경하였습니다.

Related to #50

kelvin-chihyun committed:
✨ [Feat] : 태그 캔버스 기능 추가

홈(메인) 컴포넌트 내 태그 캔버스 기능 초안입니다.

Related to #51

dani-day commented on src/components/Home/Keywords.jsx:
P3: 구 이미지를 피그마 내에서 만들면서 blur 처리를 직접 적용하다보니 옮기면서 적용이 안되어있는 것 같습니다. 기능이나 레이아웃 문제가 아니라 추후에 시간이 가능하신 경우 반영해주시면 좋을 것 같습니다 :)

<img width="1456" alt="스크린샷 2023-12-15 오후 5 29 17" src="https://github.com/ReOpenSG/ReOpenSG/assets/134567470/148ffcb8-2b23-4dc0-9575-4b8044cc416e">

kelvin-chihyun replied:
해결했습니당!

dani-day commented on src/components/Home/Industries.jsx:
P1: Industries 의 디자인 시안이 변경되면서 레이아웃을 중앙정렬로 변경하였으므로, 이 부분 반영해주시면 좋을 것 같습니다. Swiper로 적용할 시에는 붙어있거나 가려지는 부분이 있더라도 문제가 없고, 검정 박스 안에서 이어지는 느낌을 의도하기 위한 레이아웃이었으나 현재는 이미지 배치 후 hover 하는 방식이라 중앙 정렬이 더 자연스럽지 않을까 싶습니다.

> 현 레이아웃
![image](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/640740aa-0c62-4c42-814f-a3a8e33db5c7)

> 수정 요청 시안
<img width="438" alt="image" src="https://github.com/ReOpenSG/ReOpenSG/assets/134567470/b380589c-6ba9-4048-b335-1962fb4fc139">

kelvin-chihyun replied:
넵 수정했습니다!

dani-day commented on src/components/Home/Industries.jsx:
```diff
@@ -21,7 +57,26 @@ function Industries() {
+          {industryData.map((item) => (
+            <div
+              key={item.id}
+              className="w-full max-w-[705px] flex flex-1 flex-col justify-center tablet:justify-normal desktop:justify-normal flex-wrap bg-cover rounded-[20px]  min-w-[343px] tablet:min-w-[315px] desktop:min-w-[315px] h-[180px] tablet:h-[360px] desktop:h-[360px] gap-open-sm opacity-80 active:opacity-100 pointerhover:hover:opacity-100 pointerhover:hover:scale-105 hover:z-10 mouse  -text--openfoun ...
```
P2: CSS 수정을 요청드리고자 합니다. 디자인 시안에서 의도했던 바가 hover 전 어두운 상태에서 hover 이후 밝아지는 것에 더 가까워서 opacity 보다는 brightness 로 적용하는 것이 어떨지 의견드립니다 :)

> opacity 적용 시
![스크린샷 2023-12-15 오후 5 23 14](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/414fc58b-c0e0-467d-ab3d-c720e2ed825a)

> brightness 적용 시
![스크린샷 2023-12-15 오후 5 22 55](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/bbd194b4-8766-4f81-bc6a-b17c9fc91eff)

kelvin-chihyun replied:
네 훨씬 좋아보입니다! 반영하였습니다.

dani-day reviewed (changes requested):
코드에 의견 달아두었습니다. 확인 후 가능하신 부분 수정해주시면 감사하겠습니다 :)

ryudg reviewed (approved):
다연님 의견 이외 의견 없습니다!!!

kelvin-chihyun committed:
✨ [Design] : 디자인 일부 수정

PR 코멘트와 디자인 시안에 따라, Industries.jsx의 opacity => brightness로 변경,
레이아웃 일부 변경하였습니다.

Related to #50

kelvin-chihyun committed:
🌈 [Design] : 디자인 일부 수정

background image가 repeat되는 현상을 발견하여,
bg-cover, bg-no-repeat 속성을 추가하였습니다.

Related to #50

kelvin-chihyun committed:
🌈 [Design] : 디자인 일부 수정

TagCanvas에 bg-image를 적용하였습니다.

Related to #51

ingbinsee commented on src/components/Home/Keywords.jsx:
```diff
@@ -0,0 +1,122 @@
+          <li>
+            <Link to="/solutions/SCS Simulator">SCS Simulator</Link>
+          </li>
+          <li>
```
P3: AGV의 4가지 타입이므로 AGV임을 명시해주면 좋을 것 같습니다.

ingbinsee reviewed (approved):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

Keywords.jsx(P3 review)

kelvin-chihyun committed:
📂 [Rename] : 파일 일부 수정

파트너사 이미지 파일을 변경하였습니다.

kelvin-chihyun committed:
➕ [Add]	: 이미지 파일 추가

메인 페이지 solution 부분에 들어가는 일러스트 이미지를 추가하였습니다.

Related to #50

kelvin-chihyun committed:
🌈 [Design] : 디자인 일부 변경

일러스트 이미지 추가에 따라 디자인을 일부 변경하였습니다.

Related to #50

kelvin-chihyun committed:
♻️ [Refactor] : 코드 리팩토링

Product Card 컴포넌트를 만들어,
기존 Products 컴포넌트 일부를 리팩토링하였습니다.

Related to #50

kelvin-chihyun committed:
🌈 [Design] : CSS 일부 변경

레이아웃의 margin이 메인페이지 타 컴포넌트와 일치하지 않아 수정하였습니다.

Related to #50

kelvin-chihyun committed:
🌈 [Design] : CSS 일부 변경

Industries 컴포넌트에 padding, margin 값을 변경하고, w-full 적용 등
일부 CSS를 변경했습니다.

Related to #50

kelvin-chihyun committed:
♻️ [Refactor] : 코드 리팩토링

CSS 우선순위에 따라 important는 지양하고,
더 낮은 우선순위의 CSS를 덮어쓰는 방식으로 수정했습니다.

kelvin-chihyun committed:
🐛 [Fix] : 용어 일부 수정

PR 코멘트에 따라
키워드에 AGV를 명시하였습니다.

kelvin-chihyun committed:
🐛 [Fix] : 용어 일부 수정

devices/Microsoft HoloLens 2  =>  devices/Microsoft 경로 변경에 따른
라우팅 변경

kelvin-chihyun committed:
🌈 [Design] : 레이아웃 일부 수정

레이아웃이 통일성 없이 구성되어있어 일부 수정하였습니다.

kelvin-chihyun committed:
✨ [Feat] : 라우팅 추가/수정

Products 컴포넌트에 라우팅을 추가하고,
Keywords에서 라우팅 변경내용을 수정하였습니다.

kelvin-chihyun committed:
🌈 [Design] : 디자인 일부 수정

레이아웃이 통일성을 갖추지 않아,
Industries, Partner 컴포넌트의 레이아웃을 수정했습니다.

dani-day commented on src/styles/index.css:
```diff
@@ -1,3 +1,7 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+.partnerSwiper .swiper-wrapper {
+  @apply relative ease-linear !important;
+}
```
P3: index.css 에 설정해두었던 스타일 필요하지 않은 경우, 삭제해주시면 좋을 것 같습니다 :)

kelvin-chihyun replied:
넵 삭제하였습니다!

dani-day reviewed (approved):
수고하셨습니다! 👍

kelvin-chihyun committed:
🗑️ [Deprecated] : 불필요 코드 제거

불필요한 코드를 제거하였습니다.
